app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -119,7 +119,11 @@ def complete_response_data
   end
 
   def invite_required?
-    SiteSetting.invite_only?
+    if SiteSetting.invite_only?
+      path = Rails.application.routes.recognize_path(@origin)
```

Comment:
Does this work for subfolder sites?

Reply:
I change this to use `Discourse.route_for` which actually uses the same API but with better error handling :)